File: VB/MyRestService/Controllers/CategoriesController.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Net.Http
Imports System.Web.Http
Imports Newtonsoft.Json

Namespace MyRestService.Controllers
    Public Class CategoriesController
        Inherits ApiController

        Private _context As TestDataEntities
        Public Sub New()
            _context = New TestDataEntities()
        End Sub
        Public Function [Get]() As IEnumerable(Of Category)
            Dim options = Request.GetQueryNameValuePairs().ToDictionary(Function(x) x.Key, Function(x) JsonConvert.DeserializeObject(x.Value)) 'parsed options

            'see the QueryHelper class for the implementation
            Dim query = _context.Categories.AsEnumerable().FilterByOptions(options).SortByOptions(options).PageByOptions(options) 'paging - sorting - filtering
            Return query
        End Function
        Public Function [Get](ByVal id As Integer) As Category
            Return _context.Categories.Find(id)
        End Function
        Public Function Post(ByVal cat As Category) As Integer
            _context.Categories.Add(cat)
            Return _context.SaveChanges()
        End Function
        Public Function Put(ByVal cat As Category) As Integer
            Dim categ As Category = _context.Categories.Find(cat.CategoryID)
            categ.CategoryName = cat.CategoryName
            Return _context.SaveChanges()
        End Function
        Public Function Delete(ByVal id As Integer) As Integer
            Dim cat As Category = _context.Categories.Find(id)
            _context.Categories.Remove(cat)
            Return _context.SaveChanges()
        End Function
    End Class
End Namespace

File: CS/MyRestService/Controllers/CategoriesController.cs
using System;
using System.Collections.Generic;
using System.Configuration;
using System.Linq;
using System.Net;
using System.Net.Http;
using System.Web.Http;

namespace MyRestService.Controllers {
    public class CategoriesController : ApiController {
        private TestDataEntities _context;
        public CategoriesController() {
            _context = new TestDataEntities();
        }
        public IEnumerable<Category> Get() {
            return _context.Categories.AsEnumerable();
        }
        public Category Get(int id) {
            return _context.Categories.Find(id);
        }
        public int Post(Category cat) {
            _context.Categories.Add(cat);
            return _context.SaveChanges();
        }
        public int Put(Category cat) {
            Category categ = _context.Categories.Find(cat.CategoryID);
            categ.CategoryName = cat.CategoryName;
            return _context.SaveChanges();
        }
        public int Delete(int id) {
            Category cat = _context.Categories.Find(id);
            _context.Categories.Remove(cat);
            return _context.SaveChanges();
        }
    }
}
